File: 083/RemoveDuplicatesfromSortedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class RemoveDuplicatesfromSortedList {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode current = head, ln = head;
        while (ln !=null) {
            if (current.val < ln.val) current = current.next = ln;
            ln = ln.next;
        }
        if (current != null) current.next = null;
        return head;
    }
}

File: 083/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
    let curr = head;
    let ln = head;
    while (ln != null) {
        if (curr.val < ln.val) {
            curr.next = ln;
            curr = curr.next;
        }
        ln = ln.next;
    }
    if (curr != null)
        curr.next = null;
    return head;
};